packages/server/src/router.ts

```diff
@@ -528,11 +571,14 @@ export class Router<
    * Merge router with other router
    * @param router
    */
-  public merge<TChildRouter extends Router<TContext, any, any, any, any, any>>(
+  public merge<
+    TChildRouter extends Router<TContext, any, any, any, any, any, any>,
```

Comment:
We don't wanna merge routers that have different meta shapes (at least for now)


```suggestion
    TChildRouter extends Router<TContext, any, TMeta, any, any, any, any>,
```